lib/teleterm/clusters/cluster.go

```diff
@@ -127,7 +127,20 @@ func (c *Cluster) GetWithDetails(ctx context.Context) (*ClusterWithDetails, erro
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		authClusterID = clusterName.GetClusterID()
+
+		// get anonymization key from license file if available, fallback to cluster ID
+		pem, err := authClient.GetLicense(ctx)
```

Comment:
This is basically never allowed unless you're the auth server, you have tinkered with undocumented resource kinds in your roles, or are connected to a teleport enterprise "dashboard" tenant. It's outright not allowed in cloud proper, for example.

We should either start returning the anonymization key as a transient field in the `ClusterName` resource (or some other resource if you really want to go that way) or maybe return it in the `PongResponse`. The former is a bit hacky but the anonymization key, while not super sensitive, should only be distributed to authenticated clients in the cluster, and I'm not sure if we treat the Ping call as such.

Reply:
I tried to move this logic into the `ClusterName` resource per your suggestion (see https://github.com/gravitational/teleport/pull/35652/commits/4ca543785f0cf31e41ca24a44c22822ccdebc84e), but failed to find a proper place to set new `AnonymizationKey` field server-side.

I guess that it is should be somewhere around `lib/auth/apiserver.go`, but stuck around this method: https://github.com/gravitational/teleport/blob/f998daed0360c3889471b7cd27f4dff587e151b3/lib/cache/cache.go#L1780

Need assistance here.

Reply:
`GetClusterName` on `*auth.Server` currently implicitly resolves to `GetClusterName` on `auth.Server.Cache`, but it can be defined as a method on `*auth.Server` that gets the value from Cache and then adds the anonymization key from the license to it - and probably also overrides "cluster ID" with it, so that clients that don't know about the new field will still use the old field to do anonymization.

Reply:
Defined `GetClusterName` on `*auth.Server`, thanks for the hint.